mailpoet/lib/Segments/DynamicSegments/FilterDataMapper.php

```diff
@@ -323,6 +324,15 @@ private function createWooCommerce(array $data): DynamicSegmentFilterData {
       $filterData['operator'] = $data['operator'];
       $filterData['payment_methods'] = $data['payment_methods'];
       $filterData['used_payment_method_days'] = intval($data['used_payment_method_days']);
+    } elseif (in_array($data['action'], WooCommerceCustomerTextField::ACTIONS)) {
+      if (empty($data['value'])) throw new InvalidFilterException('Missing value', InvalidFilterException::MISSING_VALUE);
+      if (empty($data['operator'])) throw new InvalidFilterException('Missing operator', InvalidFilterException::MISSING_VALUE);
```

Comment:
And just one more nitpick that maybe it is just a personal preference of mine but I try to avoid using ifs without brackets. If my memory is not failing me, I believe the WP coding standard doesn't allow them (I know we don't use it). I tend to find ifs without brackets a little harder to read and a little easier to introduce errors (adding a new line below the if wrongly assuming that it will be inside the if block).

Since it seems our coding standard is ok with it, feel free to ignore this comment.

Reply:
Thanks!